pydoctor/templatewriter/writer.py

```diff
@@ -1,28 +1,35 @@
 """Badly named module that contains the driving code for the rendering."""
 
-from abc import ABC
-from typing import Type
+
+from typing import Iterable, Type, Optional, List
 import os
 import shutil
+from typing import IO
+from pathlib import Path
 
+from pydoctor.templatewriter import IWriter
 from pydoctor import model
-from pydoctor.templatewriter import DOCTYPE, pages, summary
-from pydoctor.templatewriter.util import templatefile
-from twisted.python.filepath import FilePath
-from twisted.web.template import flattenString
-
+from pydoctor.templatewriter import DOCTYPE, pages, summary, TemplateLookup
+from twisted.web.template import flattenString, Element
+from twisted.python.failure import Failure
 
-def flattenToFile(fobj, page):
+def flattenToFile(fobj:IO[bytes], page:Element) -> None:
+    """
+    This method writes a page to a HTML file.
+    @raises Exception: If any failure during L{flattenString} call. 
+    """
     fobj.write(DOCTYPE)
-    err = []
-    def e(r):
-        err.append(r.value)
-    flattenString(None, page).addCallback(fobj.write).addErrback(e)
+    err: List[Failure] = []
+    flattenString(None, page).addCallback(fobj.write).addErrback(err.append)
     if err:
-        raise err[0]
+        raise err.pop().value
```

Comment:
This will raise the last error instead of the first. Why this change in behavior?

Reply:
Did not meant to change the behaviour. 
`raise err.pop(0).value` looks better then ?